### PR TITLE
@types/react-native update types for layout props

### PR DESCRIPTION
**Pull request by gyzerok**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [ ] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://facebook.github.io/react-native/docs/layout-props.html
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by gyzerok** [comment]
@minestarks, can you, please, take a look?

File: react-native/index.d.ts [hunk 1]
```diff
@@ -554,7 +554,7 @@ declare module "react" {
         flex?: number
         flexGrow?: number
         flexShrink?: number
-        flexBasis?: number
+        flexBasis?: number | string
         flexDirection?: FlexDirection
         flexWrap?: "wrap" | "nowrap"
         height?: number
```
Comment (yuit) [inline react-native/index.d.ts]: Should all these  be `number | string` 

```ts
        height?
        left?: number
        minWidth?: number
         maxWidth?: number |
         minHeight?: number
         maxHeight?: number
         margin?: number
         marginBottom?: number
         marginHorizontal?: number
         marginLeft?: number
         marginRight?: number
         marginTop?: number
         marginVertical?: number
         overflow?: "visible" | "hidden" | "scroll"
         padding?: number
         paddingBottom?: number
         paddingHorizontal?: number
         paddingLeft?: number
         paddingRight?: number
         paddingTop?: number
         paddingVertical?: number
        right?: number
         top?: number
         width?: number
```

**Comment by yuit** [comment]
Could you also add yourself to the project header as editor ?

**Comment by gyzerok** [comment]
Hello @yuit!

Sorry, it took me quite a long time to get back to this PR. I've now made changes you requested.

> Could you also add yourself to the project header as editor ?
For now I don't feel like I have enough time to review changes. However I am considering possibility and will do it as soon as feel confident that I can do a good job here.

**Comment by gyzerok** [comment]
@minestarks can you, please, help with merging so PR don't get stale and need rebase? :)

**Comment by gyzerok** [comment]
@paulvanbrenk can I request merge here as well?

**Pull request merged**